src/server/lib/oauth/client.js

```diff
@@ -232,6 +235,17 @@ async function getOAuth2 (provider, accessToken, results) {
     if (provider.id === 'twitch') {
       headers['Client-ID'] = provider.clientId
     }
+
+    // WeChat has different url params design
+    if (provider.id === 'wechat') {
+      const wechatProfileURL = new URL(url)
+
+      wechatProfileURL.searchParams.append('access_token', results.accessToken)
+      wechatProfileURL.searchParams.append('openid', results.openid)
+
+      url = wechatProfileURL.href
+    }
```

Comment:
Custom behavior like this won't be supported in the upcoming major release. Read #2411 and the corresponding RFC #1846 for context.